packages/router/src/directives/router_link.ts

```diff
@@ -320,14 +342,11 @@ export class RouterLinkWithHref implements OnChanges, OnDestroy {
    */
   @Input() relativeTo?: ActivatedRoute|null;
 
-  private commands: any[] = [];
+  private commands: any[]|null = [];
```

Comment:
Would it make sense to init it as `null` (and similarly in the `RouterLink` class above) to avoid creating an empty array unnecessarily?
```suggestion
  private commands: any[]|null = null;
```